src/um/winsock2.rs

```diff
@@ -165,30 +226,181 @@ STRUCT!{struct linger {
+pub const WSA_QOS_RECEIVERS: c_int = WSABASEERR + 1005;
+pub const WSA_QOS_SENDERS: c_int = WSABASEERR + 1006;
+pub const WSA_QOS_NO_SENDERS: c_int = WSABASEERR + 1007;
+pub const WSA_QOS_NO_RECEIVERS: c_int = WSABASEERR + 1008;
+pub const WSA_QOS_REQUEST_CONFIRMED: c_int = WSABASEERR + 1009;
+pub const WSA_QOS_ADMISSION_FAILURE: c_int = WSABASEERR + 1010;
+pub const WSA_QOS_POLICY_FAILURE: c_int = WSABASEERR + 1011;
+pub const WSA_QOS_BAD_STYLE: c_int = WSABASEERR + 1012;
+pub const WSA_QOS_BAD_OBJECT: c_int = WSABASEERR + 1013;
+pub const WSA_QOS_TRAFFIC_CTRL_ERROR: c_int = WSABASEERR + 1014;
+pub const WSA_QOS_GENERIC_ERROR: c_int = WSABASEERR + 1015;
+pub const WSA_QOS_ESERVICETYPE: c_int = WSABASEERR + 1016;
+pub const WSA_QOS_EFLOWSPEC: c_int = WSABASEERR + 1017;
+pub const WSA_QOS_EPROVSPECBUF: c_int = WSABASEERR + 1018;
+pub const WSA_QOS_EFILTERSTYLE: c_int = WSABASEERR + 1019;
+pub const WSA_QOS_EFILTERTYPE: c_int = WSABASEERR + 1020;
+pub const WSA_QOS_EFILTERCOUNT: c_int = WSABASEERR + 1021;
+pub const WSA_QOS_EOBJLENGTH: c_int = WSABASEERR + 1022;
+pub const WSA_QOS_EFLOWCOUNT: c_int = WSABASEERR + 1023;
+pub const WSA_QOS_EUNKOWNPSOBJ: c_int = WSABASEERR + 1024;
+pub const WSA_QOS_EPOLICYOBJ: c_int = WSABASEERR + 1025;
+pub const WSA_QOS_EFLOWDESC: c_int = WSABASEERR + 1026;
+pub const WSA_QOS_EPSFLOWSPEC: c_int = WSABASEERR + 1027;
+pub const WSA_QOS_EPSFILTERSPEC: c_int = WSABASEERR + 1028;
+pub const WSA_QOS_ESDMODEOBJ: c_int = WSABASEERR + 1029;
+pub const WSA_QOS_ESHAPERATEOBJ: c_int = WSABASEERR + 1030;
+pub const WSA_QOS_RESERVED_PETYPE: c_int = WSABASEERR + 1031;
+#[inline]
+pub unsafe fn h_errno() -> c_int {
+    WSAGetLastError()
+}
+pub const HOST_NOT_FOUND: c_int = WSAHOST_NOT_FOUND;
+pub const TRY_AGAIN: c_int = WSATRY_AGAIN;
+pub const NO_RECOVERY: c_int = WSANO_RECOVERY;
+pub const NO_DATA: c_int = WSANO_DATA;
+pub const WSANO_ADDRESS: c_int = WSANO_DATA;
+pub const NO_ADDRESS: c_int = WSANO_ADDRESS;
 pub type WSAEVENT = HANDLE;
 pub type LPWSAEVENT = LPHANDLE;
 pub type WSAOVERLAPPED = OVERLAPPED;
 pub type LPWSAOVERLAPPED = *mut OVERLAPPED;
-pub const WSA_IO_PENDING: DWORD = ERROR_IO_PENDING;
-pub const WSA_IO_INCOMPLETE: DWORD = ERROR_IO_INCOMPLETE;
-pub const WSA_INVALID_HANDLE: DWORD = ERROR_INVALID_HANDLE;
-pub const WSA_INVALID_PARAMETER: DWORD = ERROR_INVALID_PARAMETER;
-pub const WSA_NOT_ENOUGH_MEMORY: DWORD = ERROR_NOT_ENOUGH_MEMORY;
-pub const WSA_OPERATION_ABORTED: DWORD = ERROR_OPERATION_ABORTED;
+pub const WSA_IO_PENDING: c_int = ERROR_IO_PENDING as i32;
+pub const WSA_IO_INCOMPLETE: c_int = ERROR_IO_INCOMPLETE as i32;
+pub const WSA_INVALID_HANDLE: c_int = ERROR_INVALID_HANDLE as i32;
+pub const WSA_INVALID_PARAMETER: c_int = ERROR_INVALID_PARAMETER as i32;
+pub const WSA_NOT_ENOUGH_MEMORY: c_int = ERROR_NOT_ENOUGH_MEMORY as i32;
+pub const WSA_OPERATION_ABORTED: c_int = ERROR_OPERATION_ABORTED as i32;
+pub const WSA_INVALID_EVENT: WSAEVENT = 0 as WSAEVENT;
```

Comment:
In this case, WSA_INVALID_EVENT is a pointer but it's not negative.
Should I still cast it using WSAEVENT?